src/endpoints/querier/main.go

```diff
@@ -0,0 +1,199 @@
+		},
+		[]string{"orgName"})
+
+	responseTimeGaugeVec = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Namespace: "AllEndpoints",
+			Name:      "http_response_time",
+			Help:      "HTTP response time partitioned by orgName",
+		},
+		[]string{"orgName"})
+
+	tlsVersionGaugeVec = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Namespace: "AllEndpoints",
+			Name:      "tls_version",
+			Help:      "TLS version reported in the response header partitioned by orgName",
+		},
+		[]string{"orgName"})
+
+	fhirVersionGaugeVec = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Namespace: "AllEndpoints",
+			Name:      "fhir_version",
+			Help:      "FHIR version reported in the conformance statement partitioned by orgName",
+		},
+		[]string{"orgName"})
+
+	totalUptimeChecksCounterVec = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Namespace: "AllEndpoints",
+			Name:      "total_uptime_checks",
+			Help:      "Total number of uptime checks partitioned by orgName",
+		},
+		[]string{"orgName"})
+
+	totalFailedUptimeChecksCounterVec = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Namespace: "AllEndpoints",
+			Name:      "total_failed_uptime_checks",
+			Help:      "Total number of failed uptime checks partitioned by orgName",
+		},
+		[]string{"orgName"})
+
+	prometheus.MustRegister(httpCodesGaugeVec)
+	prometheus.MustRegister(responseTimeGaugeVec)
+	prometheus.MustRegister(tlsVersionGaugeVec)
+	prometheus.MustRegister(fhirVersionGaugeVec)
+	prometheus.MustRegister(totalUptimeChecksCounterVec)
+	prometheus.MustRegister(totalFailedUptimeChecksCounterVec)
+
+}
+
+func setupServer() {
+	// Setup hosted metrics endpoint
+	http.Handle("/metrics", promhttp.Handler())
+	// TODO: Configure port in configureation file
+	var err = http.ListenAndServe(":8443", nil)
+	if err != nil {
+		// TODO: Use a logging solution instead of pringln
+		println("HTTP Request Error: ", err.Error())
+	}
+}
+
+func main() {
+
+	go setupServer()
+
+	var endpointsFile string
+	if len(os.Args) != 1 {
+		endpointsFile = os.Args[1]
+	} else {
+		println("ERROR: Missing endpoints list command-line arguemnt")
+		return
+	}
+	// Data in resources/EndpointSources was taken from https://fhirfetcher.github.io/data.json
+	var listOfEndpoints = fetcher.GetListOfEndpoints(endpointsFile)
+	initializeMetrics(listOfEndpoints)
+
+	var queryCount = 0
+	// Infinite query loop
+	for {
+		for _, endpointEntry := range listOfEndpoints.Entries {
+			// TODO: Distribute calls using a worker of some sort so that we are not sending out a million requests at once
+			var url = endpointEntry.FHIRPatientFacingURI
+			var orgName = endpointEntry.OrganizationName
+			// Long polling stats will be queried for every 6 hours
+			var longPollingInterval = (queryCount%72 == 0)
+			getHTTPRequestTimingFor(url, orgName, longPollingInterval)
+		}
+		runtime.GC()
+		// Polling interval, only necessary when running http calls asynchronously
+		// time.Sleep(5 * time.Minute)
```

Comment:
instead of 5 minutes, it is however long it takes to run through all of the queries serially. It has taken 3-5 minutes to run through all of the requests serially. So longPollingInterval still works the same way in that it will scrape the capability statement every 72nd query.

Reply:
right, but not necessarily every 6 hours.

anyways, at this point, whatever. But in the future, you should be able to specify the long poll interval and have it always be that regardless of serially/async/if short poll changes, etc.